tests/app/main/test_frameworks.py

```diff
@@ -2805,14 +2810,188 @@ def test_get_with_with_previous_answers(self, data_api_client):
+                framework_slug="g-cloud-9",
+                declaration={"status": "started",
+                             "conspiracy": False,
+                             "corruptionBribery": True,
+                             "fraudAndTheft": False,
+                             "terrorism": True,
+                             "organisedCrime": False,
+                             },
+                prefill_declaration_from_framework_slug="digital-outcomes-and-specialists-2"
+            )
+            # The previous declaration to prefill from - has relevant answers but should not ever be called
             data_api_client.get_supplier_declaration.return_value = {
-                "declaration": {"status": "started"}
+                'declaration': self.supplier_framework(
+                    framework_slug="digital-outcomes-and-specialists-2",
+                    declaration={"status": "complete",
+                                 "conspiracy": True,
+                                 "corruptionBribery": False,
+                                 "fraudAndTheft": True,
+                                 "terrorism": False,
+                                 "organisedCrime": False,
+                                 },
+                )["frameworkInterest"]["declaration"]
+            }
+
+            # The grounds-for-mandatory-exclusion section has "prefill: True" in the declaration manifest
+            res = self.client.get(
+                '/suppliers/frameworks/g-cloud-9/declaration/grounds-for-mandatory-exclusion')
+
+            assert res.status_code == 200
+            doc = html.fromstring(res.get_data(as_text=True))
+
+            # Previous framework and declaration have not been fetched
+            data_api_client.get_framework.assert_called_once_with('g-cloud-9')
+            assert data_api_client.get_supplier_declaration.called is False
+
+            # Radio buttons have been filled with the current answers; not those from previous declaration
+            assert len(doc.xpath('//input[@id="input-conspiracy-no"]/@checked')) == 1
+            assert len(doc.xpath('//input[@id="input-corruptionBribery-yes"]/@checked')) == 1
+            assert len(doc.xpath('//input[@id="input-fraudAndTheft-no"]/@checked')) == 1
+            assert len(doc.xpath('//input[@id="input-terrorism-yes"]/@checked')) == 1
+            assert len(doc.xpath('//input[@id="input-organisedCrime-no"]/@checked')) == 1
+
+            # No blue banner message is shown at top of page
+            assert len(doc.xpath('//div[@class="banner-information-without-action"]')) == 0
+
+            # No blue information messages are shown next to each question
+            info_messages = doc.xpath('//div[@class="message-wrapper"]//span[@class="message-content"]')
+            assert len(info_messages) == 0
+
+    def test_answers_not_prefilled_if_section_marked_as_prefill_false(self, data_api_client):
```

Comment:
Do we test a case where we're trying to prefill from a framework that is missing some of the desired questions?